src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts

```diff
@@ -90,7 +90,7 @@ export abstract class AbstractFileDialogService implements IFileDialogService {
 			return this.preferredHome(schemeFilter);
 		}
 
-		return resources.dirname(candidate);
+		return this.configurationService.getValue('files.dialog.currentDirectory') ? candidate : resources.dirname(candidate);
```

Comment:
Are we certain this works if the file path is not a folder path?

Reply:
I'm not entirely sure if it works for non-folder paths, as I only tested the scenarios shown in the videos. If you have the opportunity to test it further and make any necessary fixes or improvements, that would be great.

Reply:
Tried it out and it works.

Reply:
Actually, I tested it incorrectly. This does not work.

Reply:
If the setting is set, I think it's fair to do a `this.fileService.stat` and see if the file is directory.